flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/plan/nodes/logical/FlinkLogicalDataStreamTableScan.scala

```diff
@@ -51,46 +48,30 @@ class FlinkLogicalNativeTableScan(
   }
 }
 
-class FlinkLogicalNativeTableScanConverter
+class FlinkLogicalDataStreamTableScanConverter
   extends ConverterRule(
     classOf[LogicalTableScan],
     Convention.NONE,
     FlinkConventions.LOGICAL,
-    "FlinkLogicalNativeTableScanConverter") {
+    "FlinkLogicalDataStreamTableScanConverter") {
 
   override def matches(call: RelOptRuleCall): Boolean = {
     val scan: TableScan = call.rel(0)
-    FlinkLogicalNativeTableScan.isLogicalNativeTableScan(scan)
+    val dataStreamTable = scan.getTable.unwrap(classOf[DataStreamTable[_]])
+    dataStreamTable != null
   }
 
   def convert(rel: RelNode): RelNode = {
     val scan = rel.asInstanceOf[TableScan]
-    FlinkLogicalNativeTableScan.create(rel.getCluster, scan.getTable)
+    val traitSet = rel.getTraitSet.replace(FlinkConventions.LOGICAL)
+    new FlinkLogicalDataStreamTableScan(
+      rel.getCluster,
+      traitSet,
+      scan.getTable
+    )
   }
 }
 
-object FlinkLogicalNativeTableScan {
-  val CONVERTER = new FlinkLogicalNativeTableScanConverter
-
-  def isLogicalNativeTableScan(scan: TableScan): Boolean = {
-    val dataStreamTable = scan.getTable.unwrap(classOf[DataStreamTable[_]])
-    dataStreamTable != null
-  }
-
-  def create(cluster: RelOptCluster, relOptTable: RelOptTable): FlinkLogicalNativeTableScan = {
-    val table = relOptTable.unwrap(classOf[Table])
-    val traitSet = cluster.traitSetOf(Convention.NONE).replaceIfs(
-      RelCollationTraitDef.INSTANCE, new Supplier[util.List[RelCollation]]() {
-        def get: util.List[RelCollation] = {
-          if (table != null) {
-            table.getStatistic.getCollations
-          } else {
-            ImmutableList.of[RelCollation]
-          }
-        }
-      })
-    val scan = new FlinkLogicalNativeTableScan(cluster, traitSet, relOptTable)
-    val newTraitSet = scan.getTraitSet.replace(FlinkConventions.LOGICAL).simplify()
-    scan.copy(newTraitSet, scan.getInputs).asInstanceOf[FlinkLogicalNativeTableScan]
-  }
```

Comment:
The implementation looks a bit weird, so I revert this and add this back when need.